### PR TITLE
Display errors in user friendly way

### PR DESCRIPTION
**Pull request by kasperski95**
Based on https://github.com/software-mansion/protostar/pull/41

- added `--no-color` parameter
- added `ProtostartException`
- updated help descriptions

### CODE REVIEW

File: src/utils/retrieve_real_package_name_test.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+@pytest.fixture
+def package_dir(packages_dir: str, package_name: str):
+    package_dir = path.join(packages_dir, package_name)
+    mkdir(package_dir)
+    return package_dir
+
+
+# Patching doesn't work for some reason
```
Comment (Arcticae) [inline src/utils/retrieve_real_package_name_test.py:33]: This should be removed?

File: src/utils/standard_log_formatter.py [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
-    if level == "ERROR":
-        return Fore.RED
-    return Fore.RESET
+class StandardLogFormatter(Formatter):
+    def __init__(self, log_color_provider: LogColorProvider) -> None:
+        super().__init__()
+        self._log_color_provider = log_color_provider
 
+    def get_level_color(self, level: str) -> str:
+        if level == "DEBUG":
```
Comment (Arcticae) [inline src/utils/standard_log_formatter.py:13]: can we do a mapping here?

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: src/commands/remove/remove_package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        submodule.remove(force=True)
+
+    except InvalidGitRepositoryError as _err:
+        raise removal_exceptions.InvalidLocalRepository(
+            """Git repository must be initialized in order to remove packages.
+Run `protostar init` or `git init` to create a git repository."""
+        )
+    except (ValueError, IndexError, AttributeError) as _err:
+        raise removal_exceptions.PackageNotFound(
+            f"Protostar couldn't find the following package: {package_name}"
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/commands/remove/remove_package.py:20]: Other package managers informs a user that a package couldn't be found. 

- Poetry: `"Package fjsklfjdslk not found"`
- Yarn: `This module isn't specified in a package.json file.`
- Foundry: `error: `pathspec 'lib/bar' did not match any file(s) known to git`
- Bash: `rm ./foo`: No such file or directory

File: src/commands/install/install_package_from_repo.py [hunk 4]
```diff
@@ -22,21 +22,21 @@ def install_package_from_repo(
     try:
         repo = Repo(repo_root_dir)
     except InvalidGitRepositoryError as _err:
-        raise installation_exceptions.InvalidLocalRepository()
-    except NoSuchPathError as _err:
-        raise installation_exceptions.InvalidLocalRepository()
+        raise installation_exceptions.InvalidLocalRepository(
+            """Git repository must be initialized in order to install packages.
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/commands/install/install_package_from_repo.py]: A user may also try to install packages without initializing the project.

**Review by Arcticae — APPROVED** [review]

**Pull request merged**